sw/device/tests/otbn_isa_test.c

```diff
@@ -0,0 +1,144 @@
+    [8] = {0xac896525, 0x679944c4, 0x9641a791, 0x386507da, 0x77830eb1,
+           0x76364ab0, 0xddd71629, 0x870333f9},
+    [9] = {0xcccccd55, 0x555554cc, 0xcccccd55, 0x555554cc, 0xb4d6d555,
+           0x35d9da9b, 0xf2c374c3, 0xd7c12b4d},
+    [10] = {0xf0cd30f0, 0x45114443, 0xd30cf2cd, 0x1045054f, 0x32ced2ed,
+            0x54144010, 0x1112d2ed, 0x05011151},
+    [11] = {0xbbbc3433, 0x77dd55d5, 0xc334b4c4, 0x7d7557df, 0x44b43bc4,
+            0x77775ff5, 0xccc4433c, 0xd75777fd},
+    [12] = {0x22229a9a, 0xcd32ab2b, 0x299b1b2a, 0xd2caad35, 0xab1aa22a,
+            0xccccb54a, 0x332aa9a2, 0x2caccd53},
+    [13] = {0x97ba66a7, 0x20896565, 0xa689a3aa, 0x4a25a045, 0x43c8b58a,
+            0x1252555a, 0x5564a69a, 0xa1a55408},
+    [14] = {0x7956327a, 0xbcee9991, 0x630e74e6, 0x8dba5ca7, 0x2c59e406,
+            0xac10254c, 0xd09a8aec, 0x5ec45f47},
+    [15] = {0xac896524, 0xbcee9a19, 0x9641a791, 0x8dba5d2f, 0x77830eb1,
+            0xcb8ba005, 0xddd71629, 0xdc58894e},
+    [16] = {0xb9dd0141, 0xedbc1090, 0xd024bed4, 0x1cf04d7a, 0xeee357b5,
+            0xdf1f0aa4, 0x888f503c, 0xce52215b},
+    [17] = {0x33333331, 0x55555555, 0x33333333, 0x55555555, 0x33333333,
+            0x55555555, 0x33333333, 0x55555555},
+    [18] = {0x53769ada, 0x9866bb3b, 0x69be586e, 0xc79af825, 0x22168a4e,
+            0x34745fe9, 0xbbc28381, 0x23a7769f},
+    [19] = {0x00000000, 0x00000000, 0x09981800, 0x828aa801, 0x8818910a,
+            0x8888a009, 0x00088982, 0x28a88800},
+    [20] = {0xb9dd0130, 0xedbc10a1, 0x69be57c3, 0xc79af825, 0x887cf14e,
+            0x89c9b54f, 0x2228e9d6, 0x78fccc06},
+    [21] = {0xdbff9bfa, 0xbaeebbbb, 0xf9bfd9ee, 0xefbafabd, 0x887cf1ee,
+            0x89c9b54f, 0x2228e9d6, 0x78fccc06},
+    [22] = {0xdbff9db7, 0xbaeebbbb, 0xf9bfd9ee, 0xefbafabd, 0x887cf1ee,
+            0x89c9b54f, 0x2228e9d6, 0x78fccc06},
+    [23] = {0xdbff99f3, 0xbaeebbbb, 0xf9bfd9ee, 0xefbafabd, 0x887cf1ee,
+            0x89c9b54f, 0x2228e9d6, 0x78fccc06},
+    [24] = {0x1234abcd, 0xd0beb533, 0xcafed00d, 0xdeadbeef, 0xfacefeed,
+            0xaaaaaaaa, 0xbbbbbbbb, 0xcccccccc},
+    [25] = {0x1234abcd, 0xd0beb533, 0xcafed00d, 0xdeadbeef, 0xfacefeed,
+            0xaaaaaaaa, 0xbbbbbbbb, 0xcccccccc},
+    [26] = {0xdbff9bfa, 0xbaeebbbb, 0xf9bfd9ee, 0xefbafabd, 0x887cf1ee,
+            0x89c9b54f, 0x2228e9d6, 0x78fccc06},
+    [27] = {0x11109898, 0x8822aa2a, 0x09981808, 0x828aa820, 0x88189108,
+            0x8888a00a, 0x00088990, 0x28a88802},
+    [28] = {0xcaef0343, 0x32cc1191, 0xf027c1f7, 0x6d30528f, 0x11e568d7,
+            0x23631fe5, 0xbbb1704f, 0xd25666ac},
+    [29] = {0xeb0953c2, 0xe0654fef, 0x63388709, 0x5763bcdf, 0x26628bdb,
+            0x64341d3c, 0x9f24f0c1, 0x4f0d4b81},
+    [30] = {0x68ba2fa1, 0xb55098e0, 0x4efa2ec9, 0xaee49292, 0xab123192,
+            0xffa3d88b, 0xe9ee7ac7, 0x2167f87d},
+    [31] = {0x0f09b7c8, 0x25769434, 0x6978ad1b, 0x67a8c221, 0x5466a52c,
+            0x73880075, 0xf9dbff5e, 0x37adadae},
+};
+
+bool test_main(void) {
```

Comment:
You could also maybe check the instruction count and error bit registers with `dif_otbn_get_insn_cnt` and `dif_otbn_get_err_bits`.

Reply:
I've added a instruction count check. The error bits should already be checked by `otbn_testutils_wait_for_done`.